.github/workflows/ci-latest-release.yml

```
@@ -101,8 +126,72 @@ jobs:
         with:
           platforms: linux/amd64,linux/arm64/v8
 
-      - name: Push KubeArmor images to Docker
-        run: GITHUB_SHA=$GITHUB_SHA ./KubeArmor/build/push_kubearmor.sh ${{ steps.vars.outputs.tag }}
+      - name: Build KubeArmor images to Docker
+        if: ${{ github.event.inputs.trivy_scan}}
+        run: GITHUB_SHA=$GITHUB_SHA PLATFORMS="linux/amd64" ./KubeArmor/build/push_kubearmor.sh ${{ steps.vars.outputs.tag }} --load
+
+      - name: Save image to tar for Trivy Scan
+        if: ${{ github.event.inputs.trivy_scan}}
+        run: |
+          docker save -o kubearmor-init.tar kubearmor/kubearmor-init:${{ steps.vars.outputs.tag }}
+          docker save -o kubearmor-ubi.tar kubearmor/kubearmor-ubi:${{ steps.vars.outputs.tag }}
+          docker save -o kubearmor.tar kubearmor/kubearmor:${{ steps.vars.outputs.tag }}
+
+      - name: Run Trivy vulnerability scanner kubearmor-init
+        if: ${{ github.event.inputs.trivy_scan}}
+        id: scan_init
+        uses: aquasecurity/trivy-action@0.28.0
+        with:
+          input: kubearmor-init.tar
+          format: 'table'
+          exit-code: '1'
+          ignore-unfixed: true
+          vuln-type: 'os,library'
+          severity: 'CRITICAL,HIGH'
+        continue-on-error: true  
+
+      - name: Run Trivy vulnerability scanner kubearmor-ubi
+        if: ${{ github.event.inputs.trivy_scan}}
+        id: scan_ubi
+        uses: aquasecurity/trivy-action@0.28.0
+        with:
+          input: kubearmor-ubi.tar
+          format: 'table'
+          exit-code: '1'
+          ignore-unfixed: true
+          vuln-type: 'os,library'
+          severity: 'CRITICAL,HIGH'  
+        continue-on-error: true    
+
+      - name: Run Trivy vulnerability scanner kubearmor
+        if: ${{ github.event.inputs.trivy_scan}}
+        id: scan_main
+        uses: aquasecurity/trivy-action@0.28.0
+        with:
+          input: kubearmor.tar
+          format: 'table'
+          exit-code: '1'
+          ignore-unfixed: true
+          vuln-type: 'os,library'
+          severity: 'CRITICAL,HIGH'   
+        continue-on-error: true   
+  
+      # Set output if any scan found vulnerabilities
+      - name: Set output for vulnerability presence
+        if: ${{ github.event.inputs.trivy_scan}}
+        id: set_output
+        run: |
+          echo "has_vulns=false" >> $GITHUB_ENV  # assume clean
+          if [[ "${{ steps.scan_init.outcome }}" == "failure" ]] || \
+            [[ "${{ steps.scan_ubi.outcome }}" == "failure" ]] || \
+            [[ "${{ steps.scan_main.outcome }}" == "failure" ]]; then
+            echo "has_vulns=true" >> $GITHUB_ENV  # block build if any failed
+          fi
+        shell: bash
+
+      - name: Build KubeArmor images to Docker
+        if: ${{ env.has_vulns == 'flase' || github.event.inputs.trivy_scan == 'false' }}
```

```suggestion
        if: ${{ github.event.inputs.trivy_scan == 'false' || env.has_vulns == 'false' }}
```

there's a typo and also `has_vulns` will be present only if `trivy_scan: true`, not sure if it's an issue but we should avoid.

---

fixed the typo and default env add for `has_vulns` to avoid the issue